rules/Symfony73/Rector/Class_/CommandHelpToAttributeRector.php

```diff
@@ -0,0 +1,209 @@
+use Rector\Symfony\Enum\SymfonyAttribute;
+use Rector\Symfony\Enum\SymfonyClass;
+use Rector\ValueObject\PhpVersionFeature;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see https://symfony.com/doc/current/console.html#help-message
+ *
+ * @see \Rector\Symfony\Tests\Symfony73\Rector\Class_\CommandHelpToAttributeRector\CommandHelpToAttributeRectorTest
+ */
+final class CommandHelpToAttributeRector extends AbstractRector implements MinPhpVersionInterface
+{
+    public function __construct(
+        private readonly PhpAttributeGroupFactory $phpAttributeGroupFactory,
+        private readonly ReflectionProvider $reflectionProvider,
+    ) {
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersionFeature::ATTRIBUTES;
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Moves $this->setHelp() to the "help" named argument of #[AsCommand]',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+use Symfony\Component\Console\Command\Command;
+
+final class SomeCommand extends Command
+{
+    protected function configure(): void
+    {
+        $this->setHelp('Some help text');
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+
+#[AsCommand(name: 'app:some', help: '<<<TXT
+Some help text
+TXT')]
+final class SomeCommand extends Command
+{
+}
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($node->isAbstract()) {
+            return null;
+        }
+
+        if (! $this->reflectionProvider->hasClass(SymfonyAttribute::AS_COMMAND)) {
+            return null;
+        }
+
+        if (! $this->isObjectType($node, new ObjectType(SymfonyClass::COMMAND))) {
+            return null;
+        }
+
+        $asCommandAttribute = $this->ensureAsCommandAttribute($node);
+
+        foreach ($asCommandAttribute->args as $arg) {
+            if ($arg->name?->toString() === 'help') {
+                return $node;
+            }
+        }
+
+        $configureClassMethod = $node->getMethod(CommandMethodName::CONFIGURE);
+        if (! $configureClassMethod instanceof ClassMethod) {
+            return null;
+        }
+
+        $helpExpr = $this->findAndRemoveSetHelpExpr($configureClassMethod);
+        if (! $helpExpr instanceof String_) {
+            return null;
+        }
+
+        $wrappedHelp = new String_(
+            "<<<TXT\n" . $helpExpr->value . "\nTXT"
```

Comment:
I think you need to use  attribute KIND NOWDOC,  see

https://github.com/nikic/PHP-Parser/blob/c9d0b6c9f2c2c323661d31f7c81a26bf479ae3ff/lib/PhpParser/Node/Scalar/String_.php#L13

something like;

```php
$wrappedHelp = new String_( $helpExpr->value, [Attributekey::KIND => String_::KIND_NOWDOC]) 
```

Reply:
I updated my code to build the `String()` like this but i got `TXT)]` in the end as there is no support in adding the new line after `TXT` is that okay? I think that is also a formatting thing and rector doesn't do that right

Reply:
symfony 7.3 require php 8.2 so I think it is fine 👍